x/btcstaking/keeper/keeper.go

```diff
@@ -84,8 +66,6 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 func (k Keeper) BeginBlocker(ctx context.Context) error {
 	// index BTC height at the current height
 	k.IndexBTCHeight(ctx)
```

Comment:
as both `k.IndexBTCHeight(ctx)` and `k.UpdatePowerDist(ctx)` are happening in `BeginBlocker`, it means in our v0.9.0 chain there is already data created and saved to database due to its usage. 

This will probably means we require some data migration during upgrade

Reply:
Compared to TGE chain, not really actually. The `IndexBTCHeight` is same in TGE chain and this version. The only difference is the voting power dist cache. As in TGE chain the code always triggers [here](https://github.com/babylonlabs-io/babylon/blob/009cd29a425d85faf8f6dae0a6ecf4e375aa8211/x/btcstaking/keeper/power_dist_change.go#L33-L41), there is no voting power dist cache and voting power table recorded. Thus even after we move both KV stores to `x/finality`, the upgrade handler does not need to do any extra thing for it.

Reply:
ahh true ! so in tge chain prefix, there is no distribution caches anywhere as there is no staker/fp with voting power. thanks for explanantion 👍